Display build all test failing due to conflicting nodelabel for `m5stack_atoms3`

**Describe the bug**
The display build all test (`tests/drivers/build_all/display`) is failing for the `m5stack_atoms3/esp32s3/procpu` due to a conflicting nodelabel within the test's application overlay.

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using? `m5stack_atoms3/esp32s3/procpu` 

**To Reproduce**
Steps to reproduce the behavior:
1. `west build -p -b m5stack_atoms3/esp32s3/procpu tests/drivers/build_all/display -T drivers.display.build.default`
2. CMake fails during DTS stage

**Impact**
Blocking CI

**Logs and console output**
```
CMake Error at /home/danieldegrasse/zephyr/zephyrproject/zephyr/cmake/modules/dts.cmake:303 (execute_process):
  execute_process failed command indexes:

    1: "Child return code: 1"

Call Stack (most recent call first):
  /home/danieldegrasse/zephyr/zephyrproject/zephyr/cmake/modules/zephyr_default.cmake:133 (include)
  /home/danieldegrasse/zephyr/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:66 (include)
  /home/danieldegrasse/zephyr/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:92 (include_boilerplate)
  CMakeLists.txt:4 (find_package)
```

--- a/tests/drivers/build_all/display/app.overlay
+++ b/tests/drivers/build_all/display/app.overlay
@@ -71,7 +71,7 @@
 				height = <128>;
 			};
 
-			st7789v: st7789v@3 {
+			test_st7789v: st7789v@3 {
 				compatible = "sitronix,st7789v";
 				reg = <3>;
 				mipi-max-frequency = <25000000>;
@@ -108,7 +108,7 @@
 				busy-gpios = <&test_gpio 0 0>;
 			};
 
-			uc8176_waveshare_epaper_gdew042t2: uc8176@5 {
+			test_uc8176_waveshare_epaper_gdew042t2: uc8176@5 {
 				compatible = "ultrachip,uc8176";
 				mipi-max-frequency = <4000000>;
 				reg = <5>;

